test/test_torch.py

```diff
@@ -11168,79 +11168,86 @@ def test_signbit_non_boolean_output(self, device, dtype):
         with self.assertRaisesRegex(RuntimeError, 'does not support non-boolean outputs'):
             torch.signbit(t, out=out)
 
-    def test_logical_any(self, device):
-        x = torch.zeros([2, 3, 400], dtype=torch.uint8, device=device)
+    @dtypes(*torch.testing.get_all_dtypes(include_half=True, include_bfloat16=False,
+                                          include_bool=True, include_complex=False))
+    def test_logical_any(self, device, dtype):
```

Comment:
instead of 2 separate tests for `test_logical_any` and `test_logical_all` refactor them to use the same helper, do a loop over dimensions and keepdim=True/False, and compare with corresponding numpy results instead of hardcoding your own "golden"results. 
It looks like there are "compare with numpy" tests below, so are these tests needed at all?